lib/json/string_parser.nit

```diff
@@ -233,30 +233,98 @@ class JSONStringParser
 
 	# Parses and returns a Nit string from a JSON String
 	fun parse_json_string: Jsonable do
+		var src = src
 		var ln = src.length
 		var p = pos
 		p += 1
 		if p > ln then return make_parse_error("Malformed JSON String")
+		var ret: nullable Buffer = null
```

Comment:
Unrelated question. Is `var ret = null` (i.e. without the explicit type) correctly handled by the current adaptive typing engine?